python/cudf/cudf/core/series.py

```diff
@@ -1000,7 +1000,25 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         if fname in binary_operations:
             not_reflect = self is inputs[0]
             other = inputs[not_reflect]
-            op = f"__{'' if not_reflect else 'r'}{binary_operations[fname]}__"
+
+            # These operators need to be mapped to their inverses when
+            # performing a reflected operation because no reflected version of
+            # the operators themselves exist.
+            ops_without_reflection = {
+                "gt": "lt",
+                "ge": "le",
+                "lt": "gt",
+                "le": "ge",
+                # ne and eq are symmetric
+                "ne": "ne",
+                "eq": "eq",
+            }
+
+            op = binary_operations[fname]
+            if op in ops_without_reflection and not not_reflect:
+                op = ops_without_reflection[op]
```

Comment:
```suggestion
                op = ops_without_reflection.get(op, op)
```
Can we remove `eq` and `ne` from the dict of special operations without reflection, and use this to default back to `op`? Their symmetry makes them unlike the others as far as I can tell.

Reply:
It does, and I considered this option but chose to just include them in the dictionary because I think it helps to document what's going on. They are still ops without reflection so they belong in the set of elements in the dictionary, it's just that they are their own inverse operators so they map back to themselves.

Reply:
Gotcha. That’s a good call, I agree with what you’re saying now that I understand it better.